src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -148,6 +146,8 @@ public ProNonCombatMoveAI(final ProAI ai) {
           break;
         }
       }
+    } else {
```

Comment:
Why does this work? The commit message has no details on why this is corrects the problem, nor what the problem was. How can one know that this isn't a problem elsewhere, that there aren't other if/else branches where this code should be called?